app/controllers/hiring_staff/organisations/schools_controller.rb

```diff
@@ -0,0 +1,29 @@
+class HiringStaff::Organisations::SchoolsController < HiringStaff::BaseController
+  before_action :set_organisation, only: [:edit, :update]
+
+  def index
+  end
+
+  def edit
+  end
+
+  def update
+    if @organisation.update(description: description)
+      redirect_to organisation_schools_path,
+        success: I18n.t('messages.organisation.description_updated', organisation: @organisation.name)
+    else
+      render :edit
+    end
+  end
+
+  private
+
+  def set_organisation
+    @organisation = params[:school_group] ? SchoolGroup.find(params[:id]) : School.find(params[:id])
```

Comment:
Can you just use `current_organisation` from `AuthenticationConcerns.rb` instead?

Reply:
I use current_organisation in the `index` method. In `edit` and `update` methods I am working with either a `school` or a `school_group` depending if a pass the parameter `school_group`. This does not depend on the session

Reply:
Yes this makes sense now I remember what's going on here!